### ISSUE TITLE
Asus C201 Support

### ISSUE THREAD

**Issue by dimkr** [puppylinux-woof-CE/woof-CE#1847]
These small and light ARMv7 Chromebooks no longer receive updates from Google, and they're perfectly capable of running Puppy.

I got an old 2GB model to boot Puppy, using https://github.com/dimkr/woof-CE-c201.

Wi-Fi with a AR9271 USB dongle works fine, audio doesn't (don't know why) and persistent installation shouldn't be a problem, because a 2GB image can easily fit a 16GB image that can be written to the eMMC using dd (see https://github.com/dimkr/devsus#usage).

For some these laptops are junk, but they're also a resource. They can be donated to disadvantaged children that lost access to education due to the pandemic, unlike students that have a computer and take advantage of remote learning.

Duplicate of #605.

**Comment by 01micko** [comment]
Hi @dimkr .. long time no see.

Now that I have some experience developing [raspup](http://raspup.eezy.xyz) and [rasvuan](http://01micko.com/13-Rasvuan-Puppy-Alpha-Release) and built many packages the infrastructure of woof is more likely to be more accommodating for other `arm` processor projects.  Unfortunately, raspberry pi boards are the opposite of 'free' thanks to the choice of the board devs choosing broadcom SOC.

It would be nice to get my hands on a c201.. we'll see.

**Comment by dimkr** [comment]
Actually, I have another Chromebook, the one I was talking about in https://medium.com/@dima_8716/go-development-on-a-99-arm-laptop-26e32d27670b. As far as I remember, it should receive updates until 2025 and it shouldn't be very hard to add support for it (plus all other models that use the same SoC) too, because most of the work is getting the kernel to build (and multiple projects already rebuild the Chrome OS kernel - Arch Linux ARM for example) and the creation of a bootable image.

**Comment by 01micko** [comment]
Ok, so there are a number of arm chromebooks http://www.linuxmadesimple.info/2019/08/all-chromebooks-with-arm-processors-in.html. That's just a 2020 list.

This hisense one looks interesting .. https://archlinuxarm.org/platforms/armv7/rockchip/hisense-chromebook-c11

So.. armhf or aarch64? That is the question

If I got hold of one I'd have to source it overseas, too expensive here.

**Comment by dimkr** [comment]
The MT8173C in my 100e is technically aarch64, but the Chrome OS it uses is a combination of a 64-bit kernel with 32-bit userspace (what Debian calls "armhf"). With soldered 4 GB of RAM, this makes sense. It receives updates until 2025 (probably because this SoC is popular in models targeted at the education market) but the GPU is not freedom-friendly.

RK3288 based Chromebooks are getting phased out, although this SoC is present in many devices including some Raspberry Pi style boards. They're ARMv7 (pure 32 bit0 but the availability of Panfrost (https://panfrost.freedesktop.org/) makes them super interesting IMHO, freedom-wise.

RK3399 is still used pretty widely (including in the Pinebook Pro!) and I think it's a good investment. Looks like Chrome OS will receive updates at least until 2023, so that means 3 years of bug and security fixes in the kernel left. Panfrost works on these too.

**Comment by dimkr** [comment]
I got it to boot, but it's a "full installation" because woof-CE, init scripts, etc' all assume it's either x86 and any installation method, or ARM and a "full installation" on a SD card.

A "frugal" installation would be more convenient on Chromebooks, because it's easier to repair, and makes it easier to free space - these machines have a 16 GB SSD, and it's very little.

I started working automating the kernel build using kernel-kit, then packing vmlinuz+initrd.gz+kernel command line during 3builddistro (rather than, at kernel compilation time, because we don't have initrd.gz there) to allow "frugal" boot with a save file.

Once I get the kernel to build, I'll adapt 3builddistro so it produces flash drive/SD card/etc' images that contain the kernel (with initrd inside it) and the Puppy SFS.
